common/wuanalysis/anarule.cpp

```diff
@@ -165,11 +165,40 @@ class IoSkewRule : public AActivityRule
     const char * category;
 };
 
+class keyedJoinExcessRejectedRowsRule : public ActivityKindRule
+{
+public:
+    keyedJoinExcessRejectedRowsRule() : ActivityKindRule(TAKkeyedjoin) {}
+
+    virtual bool check(PerformanceIssue & result, IWuActivity & activity, const WuAnalyseOptions & options) override
+    {
+        stat_type preFiltered = activity.getStatRaw(StNumPreFiltered);
+        if (preFiltered)
+        {
+            IWuEdge * inputEdge = activity.queryInput(0);
+            stat_type rowscnt = inputEdge->getStatRaw(StNumRowsProcessed);
+            stat_type preFilteredPer = (preFiltered*100/ rowscnt); // prefiltered %
+            if (  preFilteredPer > preFilteredThreshold)
+            {
+                IWuActivity * inputActivity = inputEdge->querySource();
+                stat_type timeAvgLocalExecute = inputActivity->getStatRaw(StTimeLocalExecute, StAvgX);
```

Comment:
why the time of the input activity?  That may be true some of the time, but hard to tell.  A comment should explain why.